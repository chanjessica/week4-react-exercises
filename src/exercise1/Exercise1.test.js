import React from 'react';
import {shallow} from 'enzyme';
import Exercise1 from './Exercise1';
import CoffeeCard from './CoffeeCard';

describe('Exercise1', () => {
    const wrapper = shallow(<Exercise1 />);
    const coffeeCards = wrapper.find('CoffeeCard');

    it('renders and matches snapshot', () => {
        const coffeeCard = shallow(
            <CoffeeCard
                drinkName="latte"
                price={4.00}
            />
        );
        expect(coffeeCard).toMatchSnapshot();
    });

    it('should render four CoffeeCard components', () => {
        expect(coffeeCards).toHaveLength(4);
    });

    it('should have a CoffeeCard with each drink name', () => {
        const drinkNames = coffeeCards
            .map(card => card.prop('drinkName'))
            .sort();

        expect(drinkNames).toEqual(['americano', 'espresso', 'latte', 'mocha'])
    });
});