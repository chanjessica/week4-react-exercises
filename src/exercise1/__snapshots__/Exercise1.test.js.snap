// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise1 renders and matches snapshot 1`] = `
<div
  className="CoffeeCard"
>
  <img
    alt="latte"
    className="CoffeeCard__image"
    src="https://upload.wikimedia.org/wikipedia/commons/9/9f/Latte_at_Doppio_Ristretto_Chiang_Mai_01.jpg"
  />
  <h3
    className="CoffeeCard__title"
  >
    Latte
  </h3>
  <p
    className="CoffeeCard__description"
  >
    Cost: $
    4.00
  </p>
</div>
`;
